lib/core-server/src/utils/open-in-browser.ts

```diff
@@ -1,15 +1,45 @@
 import { logger } from '@storybook/node-logger';
-import open from 'better-opn';
+import betterOpn from 'better-opn'; // betterOpn alias used because also loading open
+import open from 'open';
+import getDefaultBrowser from 'x-default-browser';
 import dedent from 'ts-dedent';
 
 export function openInBrowser(address: string) {
-  try {
-    open(address);
-  } catch (error) {
-    logger.error(dedent`
-      Could not open ${address} inside a browser. If you're running this command inside a
-      docker container or on a CI, you need to pass the '--ci' flag to prevent opening a
-      browser by default.
-    `);
-  }
+  const defaultBrowserCallback = async (err: any, res: any) => {
+    if (res.isFirefox) {
+      // Handle firefox cross-browser using open's system-specific binary utility.
+      // See https://github.com/sindresorhus/open#openapps for more information.
+      try {
+        await open(address, {
+          app: {
+            name: 'firefox',
+          },
+        });
+      } catch (error) {
```

Comment:
can we move the `try/catch` around the whole thing to remove duplicate code?

Reply:
Good call - I adjusted in the following commit.